core/services/bulletprooftxmanager/eth_confirmer.go

```diff
@@ -228,6 +235,54 @@ func (ec *ethConfirmer) CheckForReceipts(ctx context.Context, blockNum int64) er
 
 	logger.Debugw(fmt.Sprintf("EthConfirmer: fetching receipts for %v transaction attempts", len(attempts)), "blockNum", blockNum)
 
+	attemptsByAddress := make(map[gethCommon.Address][]models.EthTxAttempt)
+	for _, att := range attempts {
+		attemptsByAddress[att.EthTx.FromAddress] = append(attemptsByAddress[att.EthTx.FromAddress], att)
+	}
+
+	for from, attempts := range attemptsByAddress {
+		latestBlockNonce, err := ec.getNonceForLatestBlock(ctx, from)
+		logger.Debugw(fmt.Sprintf("EthConfirmer: There are %v attempts from address %v, and latest nonce is %v",
+			len(attempts), from, latestBlockNonce), "blockNum", blockNum)
+
+		if err != nil {
+			return errors.Wrapf(err, "unable to fetch pending nonce for address: %v", from)
+		}
+
+		likelyConfirmed := make([]models.EthTxAttempt, 0)
+		for i := 0; i < len(attempts); i++ {
+			if attempts[i].EthTx.Nonce != nil && *attempts[i].EthTx.Nonce >= int64(latestBlockNonce) {
+				likelyConfirmed = attempts[0:i]
+				attempts = attempts[i:]
```

Comment:
Let's call this something else to avoid confusion, `likelyUnconfirmed` or something